scripts/data/get_block.py

```diff
@@ -0,0 +1,155 @@
+                        pk_script: from_base16(\"{txoutput["scriptPubKey"]["hex"]}\"),
+                    }},
+                    '''
+    return payload
+
+
+def outPoint(tx):
+    return f'''
+                                OutPoint {{
+                                    txid: 0x{tx["txid"]},
+                                    vout: {str(tx["vout"])},
+                                    txo_index: 0,
+                                    amount: 0,
+                                }}'''
+
+
+def txInputCoinbase(tx):
+    witness = "0"
+    if 'txinwitness' in tx:
+        witness = tx["txinwitness"][0]
+    return f'''
+                    TxIn {{
+                        script: from_base16(\"{tx["coinbase"]}\"),
+                        sequence: {str(tx["sequence"])},
+                        witness: from_base16(\"{witness}\"),
+                        previous_output: OutPoint {{
+                            txid: 0_u256,
+                            vout: 0xffffffff_u32,
+                            txo_index: 0,
+                            amount: 0,
+                        }},
+                    }}'''
+
+
+def txInputRegular(tx):
+    witness = "0"
+    if 'txinwitness' in tx:
+        witness = tx["txinwitness"][0]
+    return f'''
+                        TxIn {{
+                            script: from_base16(\"{tx["scriptSig"]['hex']}\"),
+                            sequence: {str(tx["sequence"])},
+                            witness: from_base16(\"{witness}\"),
+                            previous_output: {outPoint(tx)},
+                        }},
+                        '''
+
+
+def txInput(tx):
+    payload = ""
+    for txinput in tx:
+        if 'coinbase' in txinput:
+            payload += txInputCoinbase(txinput)
+        else:
+            payload += txInputRegular(txinput)
+    return payload
+
+
+def checkIsSegWit(tx):
+    if not int(tx["version"]) % 2:
+        return "true"
+    return "false"
+
+
+def Txs(txsraw, blockHash, ntx):
+    payload = ''
+    verboseTx = True
+    bar = progressbar.ProgressBar(max_value=ntx)
+    for i, txid in enumerate(txsraw):
+        tx = requestRPC("getrawtransaction", [txid, verboseTx, blockHash])
+        if tx['error'] is not None:
+            print("Error recieving tx")
+            return
+        tx = tx['result']
+        payload += f'''
+            Transaction {{
+                version: {str(tx["version"])},
+                is_segwit: {checkIsSegWit(tx)},
+                lock_time: {tx["locktime"]},
+                inputs: array![{txInput(tx["vin"])}].span(),
+                outputs: array![ {txOutput(tx["vout"])}].span(),
+                }},
+                '''
+        bar.update(i)
+    return payload
+
+
+def main():
+    blockHash = ""
+    if (len(sys.argv) < 2):
+        blockHash = input("Enter block hash: ")
+    else:
+        blockHash = sys.argv[1]
+
+    block = requestRPC("getblock", [blockHash])
```

Comment:
If you set verbose = 2 (second optional parameter) getblock result will include all block transactions.